hushline/routes.py

```diff
@@ -417,16 +417,69 @@ def get_directory_users() -> list[User]:
             .all()
         )
 
+    def usersPerPage() -> int:
+        return 50
+
     @app.route("/directory")
     def directory() -> Response | str:
         logged_in = "user_id" in session
-        return render_template("directory.html", users=get_directory_users(), logged_in=logged_in)
+        isViewAllRoute = request.args.get('viewing') == 'all'
+        page = int(request.args.get('page', 0))
+        offset = page  * usersPerPage() - usersPerPage()
+        return render_template(
+            "directory.html",
+            users=get_dynamic_users(offset=offset, is_not_verified_only=isViewAllRoute),
+            logged_in=logged_in,
+            isViewAllRoute=isViewAllRoute,
+            usersPerPage=usersPerPage()
+        )
 
     @app.route("/directory/get-session-user.json")
     def session_user() -> dict[str, bool]:
         logged_in = "user_id" in session
         return {"logged_in": logged_in}
 
+    def get_dynamic_users(
+            count:int=usersPerPage(),
+            offset:int=0,
+            is_not_verified_only:bool = False
+        ) -> list[User]:
+        if is_not_verified_only:
+            return (
+                User.query.filter_by(show_in_directory=True)
+                .order_by(User.is_admin.desc(), User.display_name.asc())
+                .limit(count)
+                .offset(offset)
+                .all()
+            )
+        return (
+            User.query.filter_by(show_in_directory=True, is_verified=True)
+            .order_by(User.is_admin.desc(), User.display_name.asc())
+            .limit(count)
+            .offset(offset)
+            .all()
+        )
```

Comment:
Maybe something like this to simplify the parameter and avoid re-defining the query
```suggestion
            verified_only:bool = True
        ) -> list[User]:
        filter_by = {"show_in_directory": True}
        if verified_only:
            filter_by["verified_only"] = True
        return (
            User.query.filter_by(**filter_by)
            .order_by(User.is_admin.desc(), User.display_name.asc())
            .limit(count)
            .offset(offset)
            .all()
        )
```